src/main/java/org/s1ck/gdl/model/cnf/CNF.java

```diff
@@ -66,5 +68,32 @@ public CNF or(CNF other) {
     return this;
   }
 
+  @Override
   public String operatorName() { return "AND"; }
+
+  /**
+   * Creates a new CNF containing only predicates concerning the specified variables
+   * @param variables variables
+   * @return sub cnf
+   */
+  public CNF getSubCNF(List<String> variables) {
```

Comment:
one could use a `Set<String>` instead to avoid mistakes like ['a','a'] and consequent errors due to wrong collection sizes